### PR TITLE
Add configurations

### PR DESCRIPTION
**Pull request by EricBoersma**

### CODE REVIEW

File: lib/share_progress/configuration.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  def self.configure
+    yield configuration
+  end
+
+  class Configuration
+    attr_accessor :share_progress_uri, :share_progress_api_key
+
+    def initialize
+      @share_progress_uri = ENV['SHARE_PROGRESS_URI'] || 'run.shareprogress.org/api/v1'
+      @share_progress_api_key = ENV['SHARE_PROGRESS_API_KEY'] || 'please_set_an_api_key'
```
Comment (NealJMD) [inline lib/share_progress/configuration.rb:19]: why keep the dependency on ENV at all? what if we raise an error if the api key isn't config'd?

Comment (EricBoersma) [inline lib/share_progress/configuration.rb:19]: Backwards compatibility was my thinking.

Comment (NealJMD) [inline lib/share_progress/configuration.rb:19]: but the only people that have ever used this is us and we can change our code. let us go boldly into this brave new world!

Comment (EricBoersma) [inline lib/share_progress/configuration.rb:19]: Addendum: it also makes testing a lot easier. You can't run VCR tests correctly with an invalid key, and it's much easier to handle masked valid API keys using the environment than setting up a whole `Settings` setup which shares git ignored files just for testing. 

Is there a reason to cut the `ENV` default other than just that we don't really do most things that way anymore?

Comment (NealJMD) [inline lib/share_progress/configuration.rb:19]: OK so the ENV is the default that gets overridden? I think that's fine

**Pull request merged**